### PR TITLE
Removed modifying webhook configurations if they are not found

### PR DESCRIPTION
**Pull request by meik99**
When the operator is deployed by OLM the webhook configurations cannot be easily queried since they have semi random names.
Since certificates were generated and appended to them, the operator failed when in that case since it could not find the configurations to append them to.
With these changes it does not fail anymore if it can't find the configs, but instead just writes log messages accordingly.

Also some minor adjustments were made to the Makefile, deployments and base-CSV so the operator can more reliably be deployed via OLM

### CODE REVIEW

File: config/common/webhook/deployment-webhook.yaml [hunk 1]
```diff
@@ -57,6 +57,8 @@ spec:
               path: /healthz
               port: server-port
               scheme: HTTPS
+            initialDelaySeconds: 60
```
Comment (chrismuellner) [inline config/common/webhook/deployment-webhook.yaml:60]: this introduces a delay on all platforms. why is this needed now?

File: src/controllers/certificates/webhook_reconciler.go [hunk 2]
```diff
@@ -67,64 +62,58 @@ func (r *ReconcileWebhookCertificates) Reconcile(ctx context.Context, request re
 	r.namespace = request.Namespace
 	r.ctx = ctx
 
-	secret, err := r.getSecret()
+	mutatingWebhookConfiguration, err := r.getMutatingWebhookConfiguration(ctx)
 	if err != nil {
-		return reconcile.Result{}, err
+		log.Info("could not find mutating webhook configuration, this is normal when deployed using OLM")
```
Comment (chrismuellner) [inline src/controllers/certificates/webhook_reconciler.go:67]: shouldn't this stop the remaining reconciliation from executing?
in the case of OLM the certificates do not need to be created manually

Comment (chrismuellner) [reply src/controllers/certificates/webhook_reconciler.go:67]: the comment on the tests that we are waiting for the certificate secret would make sense here.

should we rethink if always waiting for the certificate secret before startup makes sense?

File: src/controllers/certificates/webhook_configuration_test.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
+package certificates
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/admissionregistration/v1"
```
Comment (chrismuellner) [inline src/controllers/certificates/webhook_configuration_test.go:7]:
```suggestion
	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
```

File: src/controllers/certificates/certificate_secret_test.go [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+
+import (
+	"context"
+	"testing"
+
+	"github.com/Dynatrace/dynatrace-operator/src/scheme/fake"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (chrismuellner) [inline src/controllers/certificates/certificate_secret_test.go:11]:
```suggestion
	corev1 "k8s.io/api/core/v1"
```

File: src/controllers/certificates/certificate_secret.go [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+	"bytes"
+	"context"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/Dynatrace/dynatrace-operator/src/kubeobjects"
+	"github.com/Dynatrace/dynatrace-operator/src/webhook"
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/admissionregistration/v1"
```
Comment (chrismuellner) [inline src/controllers/certificates/certificate_secret.go:13]:
```suggestion
	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
```

**Review by chrismuellner — CHANGES_REQUESTED** [review]

File: src/controllers/certificates/webhook_configuration_test.go (hunk 3 above)
Comment (meik99) [reply src/controllers/certificates/webhook_configuration_test.go]: done

File: src/controllers/certificates/certificate_secret_test.go (hunk 4 above)
Comment (meik99) [reply src/controllers/certificates/certificate_secret_test.go]: done

File: src/controllers/certificates/certificate_secret.go (hunk 5 above)
Comment (meik99) [reply src/controllers/certificates/certificate_secret.go]: done

File: config/common/webhook/deployment-webhook.yaml (hunk 1 above)
Comment (chrismuellner) [reply config/common/webhook/deployment-webhook.yaml:60]: will be removed after logic rework

File: Makefile [hunk 6]
```diff
@@ -199,10 +199,15 @@ bundle: manifests kustomize
...
-	echo 'LABEL com.redhat.delivery.backport=true' >> ./config/olm/$(PLATFORM)/bundle-$(VERSION).Dockerfile
+	echo 'LABEL com.redhat.delivery.backport=true' >> ./config/olm/$(PLATFORM)/bundle-$(VERSION).Dockerfile	
 ifeq ($(PLATFORM), openshift)
 	sed 's/\bkubectl\b/oc/g' ./config/olm/$(PLATFORM)/$(VERSION)/manifests/dynatrace-operator.v$(VERSION).clusterserviceversion.yaml > ./config/olm/$(PLATFORM)/$(VERSION)/manifests/dynatrace-operator.v$(VERSION).clusterserviceversion.yaml.output
 	mv ./config/olm/$(PLATFORM)/$(VERSION)/manifests/dynatrace-operator.v$(VERSION).clusterserviceversion.yaml.output ./config/olm/$(PLATFORM)/$(VERSION)/manifests/dynatrace-operator.v$(VERSION).clusterserviceversion.yaml
+	echo '  com.redhat.openshift.versions: v4.5-v4.9' >> ./config/olm/$(PLATFORM)/$(VERSION)/metadata/annotations.yaml
+	mv ./config/olm/$(PLATFORM)/$(VERSION)/manifests/dynatrace-operator.v$(VERSION).clusterserviceversion.yaml ./config/olm/$(PLATFORM)/$(VERSION)/manifests/dynatrace-operator.clusterserviceversion.yaml
+
+	grep -v 'scorecard' ./config/olm/$(PLATFORM)/$(VERSION)/metadata/annotations.yaml > ./config/olm/$(PLATFORM)/$(VERSION)/metadata/annotations.yaml.output
```
Comment (chrismuellner) [inline Makefile:209]: this is also relevant if `PLATFORM` is `kubernetes`.

The header `# Annotations for testing.` is also still present.

Comment (meik99) [reply Makefile]: adapted makefile

**Review by chrismuellner — APPROVED** [review]
